app/controllers/api/v1/orders_controller.rb

```diff
@@ -58,12 +58,12 @@ def show
         render json: serializer.new(@order,
           root: root,
           exclude_code_details: true,
-          include_packages: bool_param(:include_packages, true),
-          include_order: false,
+          include_packages: false,
+          include_order: true,
           include_territory: true,
           include_images: true,
           include_allowed_actions: true,
-          include_orders_packages: true,
+          include_orders_packages: false,
           exclude_stockit_set_item: true
         )
```

Comment:
Instead of putting true false explicitly. Try using app_name booleans. Such as `is_browse_app?` , `is_stock_app?`

Because this might break browse app.

cc - @namrataukirde

Reply:
Agreed that changing those is risky, good catch. However the `is_browse_app?` conditionals have their own problems and we need to phase them out gradually.

I would recommend explicit options to pass in the query. Meaning keeping parameters such as `?include_packages=false`, adding any that you might need, and having the front-end request precisely what *it needs*

#### General server design idea:

The API cannot make _assumptions_ of what the front-end needs, ever.
As a "server", it literally **"serves"** whatever we request of it (within the confines of user permissions), in a generic matter.
It also exposes actions the users are permitted to take on the data

The front-end, on the other hand, is simply a _"window"_ to look into the data, and to connect to those actions.

Designing the API as a "service" is beneficial for future-proofing. Think of the public OAuth APIs such as [Twitter API](https://developer.twitter.com/en/docs/tweets/search/api-reference/get-search-tweets), [Github API](https://developer.github.com/v3/search/#search-repositories), ... Anyone can build an app for those services. We also happen to have a few apps for the Goodcity service, but we may have more/less in the future. Endpoints should therefore be flexible to support different uses.

Sorry for the rant, it felt like an opportune moment to share 😄 

cc - @abulsayyad123 @namrataukirde @abhinav098 @steveyken

Reply:
Yes I do very much agree with this. We need to move away from things that tie our server state to the front-end. The API should never know which app it is talking too. However, there are various business logic processes that are tied just to the browse app or just to the stock app. Governing these with boolean params is a safer way to go.